.github/workflows/deployDemo.yml

```diff
@@ -23,11 +25,8 @@ jobs:
         env:
             CURL_TIMEOUT: 5
         run: make check-stg-commit check-stg-readiness
```

Comment:
Now that everything is in a pipeline I think it would be better to run `check-stg-commit` and `check-stg-readiness` at the end of the deploy `stg` job instead of the start of the `training`, `demo`, and `dev`


That way if the deploy to `stg` fails then the corresponding github action will fail

Reply:
I agree with that change. I did spend some time thinking about this and ultimately decided not to do it, to avoid introducing refactoring complexity into this change. Instead, I wanted to tackle that as part of a refactoring PR. I'm happy to add that change though.

Reply:
After thinking about this, I think I will make that suggestion part of this PR since the likelihood of prioritizing a pipeline refactor in the near future seems low.